### ISSUE TITLE
ArrayIndexOutOfBoundsException when parsing ill formed SQL ending with a comment and a semicolon

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14333]
When parsing this ill formed query:

```sql
SELECT 'b' (--;
```

Then the following exception occurs:

```
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: Index 82 out of bounds for length 82
	at org.jooq/org.jooq.impl.DefaultParseContext.peekKeyword(ParserImpl.java:13808)
	at org.jooq/org.jooq.impl.DefaultParseContext.parseQuery(ParserImpl.java:1235)
	at org.jooq/org.jooq.impl.DefaultParseContext.lambda$0(ParserImpl.java:888)
	at org.jooq/org.jooq.impl.DefaultParseContext.wrap(ParserImpl.java:14553)
	at org.jooq/org.jooq.impl.DefaultParseContext.parse(ParserImpl.java:876)
	at org.jooq/org.jooq.impl.ParserImpl.parse(ParserImpl.java:754)
	at org.jooq/org.jooq.impl.ParserImpl.parse(ParserImpl.java:749)
	at org.jooq/org.jooq.ParserCLI.render(ParserCLI.java:532)
	at org.jooq/org.jooq.ParserCLI.interactiveMode(ParserCLI.java:356)
	at org.jooq/org.jooq.ParserCLI.lambda$0(ParserCLI.java:85)
	at org.jooq/org.jooq.CLIUtil.main(CLIUtil.java:55)
	at org.jooq/org.jooq.ParserCLI.main(ParserCLI.java:75)
```

On the translator website, that just lists an unspecified `Internal Error`:

<img width="798" alt="image" src="https://user-images.githubusercontent.com/734593/205015986-3b801292-f9ca-4d6a-854d-42f08aa01713.png">

**Comment by lukaseder** [comment]
The problem is the semi colon, it seems? This parses fine:

```sql
SELECT 'b' (--
```

Producing the expected error

<img width="809" alt="image" src="https://user-images.githubusercontent.com/734593/205016628-c7b882e6-c1d8-4f08-a8ab-6dd6cf4cef93.png">

**Comment by lukaseder** [comment]
The bad syntax parses fine as well with `Parser::parseQuery`, but it fails with `Parser::parse`

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.6 (https://github.com/jOOQ/jOOQ/issues/14334), and 3.16.12 (https://github.com/jOOQ/jOOQ/issues/14335)
